generate_template.sh

```diff
@@ -23,8 +23,11 @@ rm -rf "${WORKDIR}/generatedscripts/NGS_Demultiplexing/${RAWDATANAME}/out.csv"
 ###### Dual barcode checker
 #
 sampsheet="${WORKDIR}/generatedscripts/NGS_Demultiplexing/${RAWDATANAME}/${RAWDATANAME}.csv"
-
-mac2unix "${sampsheet}"
+SAMPLESHEET_SEP=","
+cp "${sampsheet}"{,.converted} 
+sed -i 's/\r/\n/g' "${sampsheet}.converted" 
```

Comment:
```suggestion
printf '\n' > "${sampsheet}.converted" 
sed -i 's/\r/\n/g' "${sampsheet}.converted" 
```

Reply:
The updated code replaces both mac2unix as well as dos2unix resulting whatever2unix. Another typical error is a missing line end character on the last line. The suggestion above inserts a line end character before the
```sed -i "/^[\s${SAMPLESHEET_SEP}]*$/d" "${sampsheet}.converted"```
line which removes empty lines. So if it was not necessary to add another line end character the `sed` that removes empty lines will fix it.

Reply:
check, updated